roadmap/implementers-guide/src/node/backing/statement-distribution.md

```diff
@@ -1,107 +1,453 @@
+        each validator in a cluster may send to others. This bounds the number of
+        candidates.
+    -   There is a small number of validators in each group, which further limits
+        the amount of candidates.
+
+&#x2014;
+
+-   We accept candidates which don't fit in the fragment trees of any relay
+    parents.
+    -   We listen to prospective parachains subsystem to learn of new additions to
+        the fragment trees.
+    -   Use this to attempt to import the candidate later.
+
+### Grid Mode
+
+-   Every consensus session provides randomness and a fixed validator set, which
+    is used to build a redundant grid topology.
+-   This grid topology is used to create "sending" and "receiving" paths from each
+    validator group to every validator.
+-   When a node observes a candidate as backed, it sends a
+    `BackedCandidateManifest` to their "receiving" nodes.
+-   If receiving nodes don't yet know the candidate, they request it.
+-   Once they know the candidate, they respond with a
+    `BackedCandidateAcknowledgement`.
+-   Once two nodes perform a manifest/acknowledgement exchange, they can send
+    `Statement` messages directly to each other for any new statements they might
+    need.
+-   There are limitations on the number of candidates that can be advertised by
+    each peer, similar to those in the cluster. Validators do not request
+    candidates which exceed these limitations.
+-   Validators request candidates as soon as they are advertised, but do not
+    import the statements until the candidate is part of the hypothetical
+    frontier, and do not re-advertise or acknowledge until the candidate is
+    considered both backed and part of the hypothetical frontier.
+-   Note that requesting is not an implicit acknowledgement, and an explicit
+    acknowledgement must be sent upon receipt.
+
+### Old Grid Protocol
+
+-   Once the candidate is backed, wait for some time T to pass to collect
+    additional statements. Then produce a 'backed candidate packet'
+    `(CommittedCandidateReceipt, Statements)`.
+-   Members of a backing group produce an announcement of a fully-backed candidate
+    (aka "full manifest") when they are finished.
+    -   `BackedCandidateManifest`
+    -   Manifests are sent along the grid topology to peers who have the relay-parent
+        in their implicit view.
+    -   Only sent by 1st-hop nodes after downloading the backed candidate packet
+        -   TODO: why?
+    -   Ignored when received out-of-topology
+-   On every local view change, members of the backing group rebroadcast the
+    manifest for all candidates under every new relay-parent across the grid.
+-   Nodes should send a `BackedCandidateKnown(hash)` (acknowledgement)
+    notification to any peer which has sent a manifest, and the candidate has been
+    acquired by other means.
+    -   This keeps alternative paths through the topology open.
```

Comment:
this is mostly about the limitation that the runtime has no way for block authors to post statements that come 'later' and ensure those validators still get rewarded. technically, we only need enough statements to back the candidate and the manifest + request will provide that. but more statements might come shortly afterwards, and we want those to end up on-chain as well to ensure all validators in the group are rewarded.

of course, game-theoretically we'd probably expect validators to omit statements for validators they want to snub (either in the gossip layer or in the authoring layer), but that's more of a theoretical concern that would be addressed by better on-chain logic for late statements.

Reply:
Still, including later statements is also about getting more skin in the game, which is a good thing for security.

Reply:
I don't fully understand... Why do we allow for later statements if they can't be posted by the runtime?

Having alternative paths basically just maximizes the amount of statements we observe? Am I understanding correctly?

Reply:
"later" here means "after the parablock was backable off-chain but before it was posted to the chain"

We have a timeline like this:

1. candidate seconded 
2. backable in cluster
3. distributed along grid 
4. latecomers issue statements 
5. candidate posted on chain
6. really latecomers issue statements

We can make sure that the statements from (4) go on-chain, but can't do anything about (6). Note that (4) might be stuff like the 3rd backer being 100ms slower than the 2nd.

Reply:
Thanks, that makes sense now! I added a new section about this part specifically, since I feel it requires some explaining to be clear.